packages/pangea-sdk/pangea/services/vault/models/common.py

```diff
@@ -129,59 +153,59 @@ class CommonGenerateRequest(APIRequestModel):
     folder: Optional[str] = None
     metadata: Optional[Metadata] = None
     tags: Optional[Tags] = None
-    auto_rotate: Optional[bool] = None
-    rotation_policy: Optional[str] = None
-    store: Optional[bool] = None
+    rotation_frequency: Optional[str] = None
+    rotation_state: Optional[ItemVersionState] = None
     expiration: Optional[datetime.datetime] = None
-    managed: Optional[bool] = None
 
 
 class CommonGenerateResult(PangeaResponseResult):
     type: str
-    version: Optional[int] = None
-    id: Optional[str] = None
+    version: int
+    id: str
 
 
 class GetRequest(APIRequestModel):
     id: str
-    version: Optional[int] = None
+    version: Optional[Union[str, int]] = None
     verbose: Optional[bool] = None
+    version_state: Optional[ItemVersionState] = None
 
 
-class CommonGetResult(PangeaResponseResult):
-    type: str
-    id: str
+class ItemVersionData(PangeaResponseResult):
```

Comment:
Please remove public_key and secret from this object.
Also you could rename the object as CurrentVersionData.

Reply:
Aren't they returned when you do a get? they are optional values

Reply:
that's when request /list, but when request /get you also get secret or public_key. I'm using same object with optional fields

Reply:
Ok